external-health-monitor/release.sh

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2023 MinIO, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ME=$(basename "$0"); export ME
+cd "$(dirname "$0")" || exit 255
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+declare BUILD_VERSION PODMAN IMG
+BUILD_VERSION_PATTERN="^[a-zA-Z0-9_.-]+$"
+
+function print_help() {
+    cat <<EOF
+USAGE:
+  ${ME} [VERSION]
+
+EXAMPLE:
+  # Release with the latest release tag of the upstream
+  $ ${ME}
+
+  # Release the image with tag 0.10.0
+  $ ${ME} 0.10.0
+EOF
+}
+
+function init() {
+
+    if [[ $# -eq 1 ]]; then
+        case "$1" in
+            -h|--help)
+                print_help
+                exit 0
+                ;;
+            *)
+        esac
+        shift
+    elif [[ $# -gt 1 ]]; then
+        print_help
+        exit 255
+    fi
+
+
+    if  [ $# -eq 0 ]; then
+        BUILD_VERSION=$(curl -sfL https://api.github.com/repos/kubernetes-csi/external-health-monitor/releases/latest | awk '/tag_name/ { print substr($2, 3, length($2)-4) }')
+    else
+        BUILD_VERSION="${1/v/}"
+    fi
+
+    if ! [[ "${BUILD_VERSION}" =~ $BUILD_VERSION_PATTERN ]]; then
+        echo "Invalid build version: ${BUILD_VERSION}"
+        exit 255
+    fi
+
+    IMG="quay.io/praveen8/csi-external-health-monitor-controller:v${BUILD_VERSION}"
+
+    if which podman >/dev/null 2>&1; then
+        PODMAN=podman
+    elif which docker >/dev/null 2>&1; then
+        PODMAN=docker
+    else
+        echo "no podman or docker found; please install"
+        exit 255
+    fi
+
+    if "${PODMAN}" pull "${IMG}"; then
+        echo "image ${IMG} is already pushed"
+        exit 0
+    fi
+}
+
+function main() {
+    # Downloading the source tar
+    work_dir=$PWD
+    mkdir build && cd build
+    curl --silent --location --insecure --fail "https://github.com/kubernetes-csi/external-health-monitor/archive/refs/tags/v${BUILD_VERSION}.tar.gz" | tar -zxf -
+    cd external-health-monitor-"${BUILD_VERSION}"
+    make build
+    cp "$work_dir"/Dockerfile cmd/csi-external-health-monitor-controller/
+    cp LICENSE cmd/csi-external-health-monitor-controller/
+
+    # Build the image
+    "${PODMAN}" buildx build --platform linux/amd64 -f cmd/csi-external-health-monitor-controller/Dockerfile --tag "${IMG}" .
+    "${PODMAN}" push "${IMG}"
+
+    # cleanup
+    cd "$work_dir" && rm -rf build
+}
+
+init "$@"
+main "$@"
```

Comment:
1. Rename this script as `release-csi-external-health-monitor-controller.sh` and move it to project root.
2. Add below to end of `init()`
```sh
BUILD_DIR="$(mktemp -d -p "${PWD}" health-monitor-build.XXXXXXXX)"
```

3. Simplify `main()` by
```sh
curl --silent --location --insecure --fail "https://github.com/kubernetes-csi/external-health-monitor/archive/refs/tags/v${BUILD_VERSION}.tar.gz" | tar -zxf -
cd "external-health-monitor-${BUILD_VERSION}" || return
make build
"${PODMAN}" buildx build --platform linux/amd64 --tag "${IMG}" -f - . <<EOF
FROM registry.access.redhat.com/ubi8/ubi-minimal:8.8
LABEL maintainers="dev@min.io"
LABEL description="Rebuild of CSI External Health Monitor Controller for Red Hat Marketplace"
COPY ./bin/csi-external-health-monitor-controller csi-external-health-monitor-controller
COPY ./LICENSE /licenses/LICENSE
ENTRYPOINT ["/csi-external-health-monitor-controller"]
EOF
"${PODMAN}" push "${IMG}"
cd - || return
```
4. Call `main` by
```
cd "${BUILD_DIR}" || exit 255
main
cd - || exit 255
rm -fr "${BUILD_DIR}"
```
5. Remove external-health-monitor/Dockerfile file